pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -291,7 +291,17 @@ func (lb *ListenerBuilder) buildSidecarOutboundListeners(node *model.Proxy,
 ) []*listener.Listener {
 	noneMode := node.GetInterceptionMode() == model.InterceptionNone
 
-	actualWildcard, actualLocalHostAddress := getActualWildcardAndLocalHost(node)
+	oWildcardAndLocalHost := NewWildcardAndLocalHost(node.GetIPMode())
+	if oWildcardAndLocalHost == nil {
+		log.Warnf("buildSidecarOutboundListeners: Can not fetch wildcar and localhost from proxy [%s]", node.ID)
+		return nil
+	}
+	actualWildcards := oWildcardAndLocalHost.GetWildcardAddresses()
+	actualLocalHosts := oWildcardAndLocalHost.GetLocalHostAddresses()
+	if len(actualWildcards) == 0 || len(actualLocalHosts) == 0 {
```

Comment:
Same here. When will these be empty? once we detect proxy IPMode, these will have some values.

Reply:
refactor

Reply:
refactored